src/utilities/scalarValueFromAST.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+/**
+ *  Copyright (c) 2017, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import keyValMap from '../jsutils/keyValMap';
+import * as Kind from '../language/kinds';
+import type { ValueNode } from '../language/ast';
+
+/**
+ * Create a JavaScript value from a GraphQL languate AST representation
+ * of Scalar.
+ */
+export function scalarValueFromAST(astValue: ValueNode): mixed {
```

Comment:
> This looks very similar to valueFromAST - can that be used instead?

`valueFromAST` builds value based on the type definition it accepts as the second argument.
In a case of a scalar, there is no such type definition.

> If this function is for scalars, then it should likely not support LIST or OBJECT, correct?

AFAIK, the spec doesn't explicitly forbid complex scalars as LIST or OBJECT. Moreover, it's a widespread practice to create such scalars. More details in this PR:
https://github.com/facebook/graphql/pull/325